src/pk/ecc/ecc_sign_hash.c

```diff
@@ -7,43 +7,29 @@
  * guarantee it works.
  */
 
-/* Implements ECC over Z/pZ for curve y^2 = x^3 - 3x + b
- *
- * All curves taken from NIST recommendation paper of July 1999
- * Available at http://csrc.nist.gov/cryptval/dss.htm
- */
 #include "tomcrypt.h"
 
+#ifdef LTC_MECC
+
 /**
   @file ecc_sign_hash.c
   ECC Crypto, Tom St Denis
 */
 
-#ifdef LTC_MECC
-
-/**
-  Sign a hash with ECC
-  @param in       The hash to sign
-  @param inlen    The length of the hash to sign
-  @param r        The "r" integer of the signature (caller must initialize with mp_init() first)
-  @param s        The "s" integer of the signature (caller must initialize with mp_init() first)
-  @param prng     An active PRNG state
-  @param wprng    The index of the PRNG desired
-  @param key      A private ECC key
-  @return CRYPT_OK if successful
-*/
-int ecc_sign_hash_raw(const unsigned char *in,  unsigned long inlen,
-                            void *r, void *s,
-                            prng_state *prng, int wprng, ecc_key *key)
+static int _ecc_sign_hash(const unsigned char *in,  unsigned long inlen,
+                                unsigned char *out, unsigned long *outlen,
+                                prng_state *prng, int wprng, ecc_key *key, int sigformat)
 {
    ecc_key       pubkey;
-   void          *e, *p;
-   int           err;
+   void          *r, *s, *e, *p;
+   int           err, max_iterations = 20;
```

Comment:
probably we could introduce `PK_MAX_RETRIES` or something like that which can be used in all cases where we have to do retry-loops (c.f. `DH_MAKE_KEY_MAX_ITERATIONS`)? Otherwise we should use a separate one here as well.